From daea61a39b53badef423af056eddf3f29803e948 Mon Sep 17 00:00:00 2001
From: Lars Kellogg-Stedman <lars@oddbit.com>
Date: Fri, 19 Aug 2022 22:42:28 -0400
Subject: [PATCH 6/8] Fix build on mips* (inconsistent cacheflush() prototype)

Apply the main/lmdb mips* build fix to the bundled lmdb copy.
---
 libraries/liblmdb/mdb.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/libraries/liblmdb/mdb.c b/libraries/liblmdb/mdb.c
index 8cecdb2..82cefc3 100644
--- a/libraries/liblmdb/mdb.c
+++ b/libraries/liblmdb/mdb.c
@@ -77,8 +77,7 @@
 
 #if defined(__mips) && defined(__linux)
 /* MIPS has cache coherency issues, requires explicit cache control */
-#include <asm/cachectl.h>
-extern int cacheflush(char *addr, int nbytes, int cache);
+#include <sys/cachectl.h>
 #define CACHEFLUSH(addr, bytes, cache)	cacheflush(addr, bytes, cache)
 #else
 #define CACHEFLUSH(addr, bytes, cache)
-- 
2.37.1

